src/test/Recordings/Recordings.test.tsx

```diff
@@ -0,0 +1,170 @@
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import * as React from 'react';
+import renderer, { act } from 'react-test-renderer'
+import { render, screen, within } from '@testing-library/react';
+import userEvent from '@testing-library/user-event'
+import '@testing-library/jest-dom';
+import { of } from 'rxjs';
+import { ServiceContext, defaultServices } from '@app/Shared/Services/Services'
+import { Recordings } from '@app/Recordings/Recordings';
+
+jest.mock('@app/Recordings/ActiveRecordingsTable', () => {
+  return {
+    ActiveRecordingsTable: jest.fn((props) => {
+      return <div>
+                Active Recordings Table
+             </div>
+    })
+  }
+});
+
+jest.mock('@app/Recordings/ArchivedRecordingsTable', () => {
+  return {
+    ArchivedRecordingsTable: jest.fn(() => {
+      return <div>
+                Archived Recordings Table
+             </div>
+    })
+  }
+});
+
+jest.mock('@app/TargetView/TargetView', () => {
+  return {
+    TargetView: jest.fn((props) => {
+      return <div>
+                {props.pageTitle}
+                {props.children}
+             </div>
+    })
+  }
+});
+
+jest.mock('@app/Shared/Services/Api.service', () => {
+  return {
+    ApiService: jest.fn(() => {
+      return {
+        isArchiveEnabled: jest.fn()
+                          .mockReturnValueOnce(of(true))
+                          .mockReturnValueOnce(of(true))
+                          .mockReturnValueOnce(of(true))
+                          .mockReturnValueOnce(of(false))
+                          .mockReturnValue(of(true))
+      };
+    })
+  };
+});
+
+describe('<Recordings />', () => {
+  it('renders correctly', async () => {
+    let tree;
+    await act(async () => {
+      tree = renderer
+      .create(
+        <ServiceContext.Provider value = {defaultServices}>
+          <Recordings />
+        </ServiceContext.Provider>)
+    })
+    expect(tree.toJSON()).toMatchSnapshot();
+  });
+
+  it('has the correct title in the TargetView', () => {
+    render(
+      <ServiceContext.Provider value = {defaultServices}>
+				<Recordings />
+			</ServiceContext.Provider>
+		);
+
+    expect(screen.getByText('Recordings')).toBeInTheDocument();
+  })
+
+  it('handles the case where archiving is enabled', () => {
+    render(
+      <ServiceContext.Provider value = {defaultServices}>
+				<Recordings />
+			</ServiceContext.Provider>
+		);
+
+    expect(screen.getByText('Active Recordings')).toBeInTheDocument()
+		expect(screen.getByText('Archived Recordings')).toBeInTheDocument()
```

Comment:
Funny indentation here - maybe related to auto-formatting because the line above is also missing its terminating `;`?

Reply:
Hmm, weird. It looks fine in VS Code but I went ahead and added the terminating `;`, as well as removed it some places so that the `;` usage matches `About.test.tsx` which has no issues, but no luck. The formatting inside the `render` calls is also messed up but again, looks fine in VS Code. It could be an auto-formatting issue with Typescript and VS Code?

Reply:
It looks like there might be a `\t` character at the start of the line, actually. So it'll look properly indented if you have VS Code set to render those at 4 space widths or whatever.

Reply:
Nice catch, I fixed it by changing the VS Code settings to render all whitespace characters. That allowed me to delete the `\t` characters.